internal/cli/apply.go

```diff
@@ -59,12 +57,86 @@ func NewCmdApply() *cobra.Command {
 	if err != nil {
 		log.Fatalf("setting filename flag annotation: %v", err)
 	}
-	flags.BoolVarP(&o.DryRun, "dry-run", "", o.DryRun, "only print the object that would be sent, without sending it")
-	flags.BoolVarP(&o.Recursive, "recursive", "R", o.Recursive, "process the directory used in -f, --filename recursively")
+	flags.BoolVarP(&o.DryRun, "dry-run", "", o.DryRun, "Only print the object that would be sent, without sending it.")
+	flags.BoolVarP(&o.Recursive, "recursive", "R", o.Recursive, "Process the directory used in -f, --filename recursively.")
 
 	return cmd
 }
 
+func (o *ApplyOptions) Complete(cmd *cobra.Command, args []string) error {
+	return nil
+}
+
+func (o *ApplyOptions) Validate(args []string) error {
+	if len(o.Filenames) == 0 {
+		return fmt.Errorf("must specify -f FILENAME")
+	}
+	if len(args) > 0 {
+		return fmt.Errorf("unexpected arguments: %v (did you forget to quote wildcards?)", args)
+	}
+	return nil
+}
+
+func (o *ApplyOptions) Run(ctx context.Context, args []string) error {
+	client, err := client.NewFromConfigFile(defaultClientConfigFile)
+	if err != nil {
+		return fmt.Errorf("creating client: %w", err)
+	}
+
+	errs := make([]error, 0)
+	for _, filename := range o.Filenames {
+		switch {
+		case filename == "-":
+			errs = append(errs, applyFromReader(ctx, client, "<stdin>", os.Stdin, o.DryRun)...)
+		default:
+			expandedFilenames, err := expandIfFilePattern(filename)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			for _, filename := range expandedFilenames {
+				_, err := os.Stat(filename)
+				if os.IsNotExist(err) {
+					errs = append(errs, fmt.Errorf("the path %q does not exist", filename))
```

Comment:
My only question is should we fail fast here?

Reply:
IDK. Maybe? I only moved this method up without changing it. Q is whether we want to greedily apply as many resources as possible or to stop on the first error. What's K8s' behavior?

Reply:
I think this is considered a preflight check, kube bails
```
oc apply -f test.yaml,test.broke
error: the path "test.broke" does not exist
```

Reply:
```
oc apply -f test.yaml,test.broke,test.broke2
the path "test.broke" does not exist
the path "test.broke2" does not exist
```